docs/DeveloperGuide.adoc

```diff
@@ -137,193 +183,220 @@ image:BetterModelClassDiagram.png[]
+Classes used by multiple components are in the `seedu.foodiebot.commons` package.
 
-== Implementation
 
-This section describes some noteworthy details on how certain features are implemented.
+== Implementation
 
-// tag::undoredo[]
-=== [Proposed] Undo/Redo feature
+// tag::list[]
+=== [Proposed] List feature
 ==== Proposed Implementation
 
-The undo/redo mechanism is facilitated by `VersionedAddressBook`.
-It extends `AddressBook` with an undo/redo history, stored internally as an `addressBookStateList` and `currentStatePointer`.
-Additionally, it implements the following operations:
+The list command is facilitated by `ListCommandParser`.
+It extends `FoodieBotParser` to handle list canteen commands.
+It implements the following operations:
 
-* `VersionedAddressBook#commit()` -- Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` -- Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` -- Restores a previously undone address book state from its history.
+* `ListCommand#execute()` -- Updates the canteen list to show all the canteens or filtered by location.
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+[NOTE]
+`ListCommand` extends `Command` with `Command#execute()`.
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+* `ListCommandParser#arePrefixesPresent()` -- Checks if the prefixes specified in `parse()` are found in the command entered by the user.
+* `ListCommandParser#parse()` -- Identifies if prefixes have been specified and handles them respectively. +
+For example, if ListCommand has the nearest block name passed through as a parameter `list f\com1`, ListCommand will filter the canteen list with the given block name with `new ListCommand("com1")`
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
 
-image::UndoRedoState0.png[]
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
 
-image::UndoRedoState1.png[]
+The following sequence diagram shows how the list operation works:
 
-Step 3. The user executes `add n/David ...` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
 
-image::UndoRedoState2.png[]
+image::ListSequenceDiagram.png[]
```

Comment:
There shouldn't be anymore dotted lines after the "X" for listCommandParser because it is de-referenced.